<!DOCTYPE html>
<html>
<head>
  <title>From Wells to Cells</title>
  <meta charset="utf-8">
  <meta name="description" content="From Wells to Cells">
  <meta name="author" content="Rainer Machne">
  <meta name="generator" content="slidify" />
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta http-equiv="X-UA-Compatible" content="chrome=1">
  <link rel="stylesheet" href="libraries/frameworks/io2012/css/default.css" media="all" >
  <link rel="stylesheet" href="libraries/frameworks/io2012/css/phone.css" 
    media="only screen and (max-device-width: 480px)" >
  <link rel="stylesheet" href="libraries/frameworks/io2012/css/slidify.css" >
  <link rel="stylesheet" href="libraries/highlighters/highlight.js/css/tomorrow.css" />
  <base target="_blank"> <!-- This amazingness opens all links in a new tab. -->  <link rel=stylesheet href="./assets/css/my.css"></link>
<link rel=stylesheet href="./assets/css/ribbons.css"></link>

  
  <!-- Grab CDN jQuery, fall back to local if offline -->
  <script src="http://ajax.aspnetcdn.com/ajax/jQuery/jquery-1.7.min.js"></script>
  <script>window.jQuery || document.write('<script src="libraries/widgets/quiz/js/jquery.js"><\/script>')</script> 
  <script data-main="libraries/frameworks/io2012/js/slides" 
    src="libraries/frameworks/io2012/js/require-1.0.8.min.js">
  </script>
  
  

</head>
<body style="opacity: 0">
  <slides class="layout-widescreen">
    
    <!-- LOGO SLIDE -->
        <slide class="title-slide segue nobackground">
  <hgroup class="auto-fadein">
    <h1>From Wells to Cells</h1>
    <h2>Growth vs. Gene Expression in Platereader Experiments</h2>
    <p>Rainer Machne<br/></p>
  </hgroup>
  <article></article>  
</slide>
    

    <!-- SLIDES -->
    <slide class="centertext" id="slide-1" style="background:;">
  <article data-timings="">
    <style>
em {
  font-style: italic
}
</style>

<style>
oq {
    background-color:yellow;
    color:#CC2904;
    font-weight: bold;
}
</style>

<h2>Plan: Understanding Growth Curves</h2>

<table><thead>
<tr>
<th>#</th>
<th>Experiment</th>
<th>Theory</th>
</tr>
</thead><tbody>
<tr>
<td>1</td>
<td>Using R packages: <code>platexpress</code>, <code>grofit</code>, etc.</td>
<td>The Basics: What is Exponential Growth?</td>
</tr>
<tr>
<td>2</td>
<td>The Experiment: View and Analyze Data</td>
<td>Statistics: Measurement Errors</td>
</tr>
<tr>
<td>3</td>
<td>The Result: Summary Plots</td>
<td>The Model: Growth vs. Expression Rates</td>
</tr>
</tbody></table>

<p><br/></p>

<p><img src="assets/img/ecoli_20141014.png" height="250"><img src="assets/img/fritz_the_platypus.gif" height="230"><img src="assets/img/Ecoli_20161014_OD_grofit_A8.png" height="250">
<a href="https://github.com/raim/platexpress">https://github.com/raim/platexpress</a> </p>

<p><oq>Goal: learn some R for data analysis, and gain better<br/>quantitative
understanding of cell growth &amp; gene regulation</oq></p>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-2" style="background:;">
  <article data-timings="">
    
<div style='float:left;width:35%;'>
  <p><img src="assets/img/ecoli_20141014.png" height="190">
<img src="assets/img/Ecoli_20161014_OD_grofit_A8.png" height="190">
<img src="assets/img/Ecoli_20161014_OD_growthrates.png" height="190"></p>

</div>
<div style='float:right;width:60%;'>
  <ol>
<li>Use R to Analyze Growth &amp; Expression Data

<ul>
<li>Installing R packages</li>
<li>Prepare &amp; load data</li>
</ul></li>
<li>Use R <code>base</code> 

<ul>
<li>Linear regression with <code>lm(log(X) ~ time)</code></li>
<li>Non-linear fit with <code>nls(X ~ X0*exp(mu*time))</code></li>
</ul></li>
<li>Use R packages 

<ul>
<li><code>grofit</code>, <code>growthcurver</code> or <code>growthrate</code></li>
<li>Logistic, Gompertz, non-parametric growth models</li>
</ul></li>
<li>Growth vs. Gene Expression

<ul>
<li><oq>Proteins/Cell: normalize fluorescence &amp; compare</oq></li>
<li>Monod equation: growth vs. gene expression</li>
</ul></li>
</ol>

</div>
  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-3" style="background:;">
  <hgroup>
    <h3>Installing R Packages from <code>cran</code>, <code>bioconductor</code> &amp; <code>github</code></h3>
  </hgroup>
  <article data-timings="">
    <pre><code class="r">install.packages(c(&quot;grofit&quot;,&quot;growthcurver&quot;)) # at CRAN

source(&quot;https://bioconductor.org/biocLite.R&quot;) # at bioconductor
biocLite(&quot;cellGrowth&quot;)

install.packages(&quot;devtools&quot;) # R development tools
library(devtools)
install_github(&quot;raim/platexpress&quot;) # at github
</code></pre>

<p><img src="assets/img/fritz_the_platypus.gif" height="130">
<a href="https://github.com/raim/platexpress">https://github.com/raim/platexpress</a> </p>

<p><oq>Do you speak <code>git</code>? Try to install from your local copy:</oq></p>

<p><code>git clone git@github.com:raim/platexpress.git</code></p>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="centertext" id="slide-4" style="background:;">
  <hgroup>
    <h3>How to use a new R package?</h3>
  </hgroup>
  <article data-timings="">
    <pre><code class="r">## load the package &amp; explore
library(platexpress)

?platexpress # VIEW HELP FILES
vignette(&quot;platexpress&quot;) # READ THE VIGNETTE
demo(&quot;demo_ap12&quot;) # RUN THE DEMO
getData # SEE WHAT A FUNCTION DOES: just type without brackets

## APPLY TO YOUR DATA:

plate &lt;- readPlateMap(&quot;IPTG_Testreihe_3.csv&quot;)
files &lt;- c(&quot;20161201_20161201 Praktikum - pRAJ11  1_Absorbance.CSV&quot;,
           &quot;20161201_20161201 Praktikum - pRAJ11  1_Fluorescence.CSV&quot;)
raw &lt;- readPlateData(files, type=&quot;BMG&quot;)
viewPlate(raw)
</code></pre>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-5" style="background:;">
  <hgroup>
    <h3>Why Growth Rates?</h3>
  </hgroup>
  <article data-timings="">
    
<div style='float:left;width:35%;'>
  <p><img src="assets/img/ecoli_20141014.png" height="250"></p>

<ul>
<li>Injection of IPTG into one column (7 wells + 1 blank) every
100 minutes.</li>
<li>Slower response with later injections.</li>
</ul>

</div>
<div style='float:right;width:60%;'>
  <p><img src="assets/img/Ecoli_20161014_OD_growthrates.png" height="250"></p>

<p>\( \Rightarrow \) Faster growth rate \(\mu\) with later or no induction.</p>

</div>
  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-6" style="background:;">
  <hgroup>
    <h3>Why Growth Rates?</h3>
  </hgroup>
  <article data-timings="">
    
<div style='float:left;width:35%;'>
  <p><img src="assets/img/ecoli_20141014.png" height="250"></p>

<ul>
<li>Injection of IPTG into one column (7 wells + 1 blank) every
100 minutes.</li>
<li>Slower response with later injections.</li>
</ul>

</div>
<div style='float:right;width:60%;'>
  <p><img src="assets/img/Ecoli_20161014_OD_growthrates.png" height="250"></p>

<p>\( \Rightarrow \) Faster growth rate \(\mu\) with later or no induction.</p>

<p><oq>\(\mu = k \frac{\text{ribosomes}}{\text{proteins}}\)</oq></p>

<p><div  style='text-align: left;line-height: 90%;'><font size=3> 
Schaechter, Maaloe &amp; Kjeldgaard, J Gen Microbiol <b> 1958</b>: <em>Dependency on medium and temperature of cell size and chemical composition during balanced growth of <em>Salmonella typhimurium</em>.</em><br/>
<b>Koch, Can J Microbiol 1988: <em>Why can&#39;t a cell grow infinitely fast?</em></b></br>
Neidhardt, J Bacteriol 1999: <em>Bacterial growth: constant obsession with dN/dt.</em>
</font></p>

</div>
  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-7" style="background:;">
  <hgroup>
    <h3>Why Growth Rates?</h3>
  </hgroup>
  <article data-timings="">
    
<div style='float:left;width:35%;'>
  <p><img src="assets/img/ecoli_20141014.png" height="250"></p>

<p><img src="assets/img/scott14_fig1b.png" height="220"></p>

</div>
<div style='float:right;width:60%;'>
  <p><img src="assets/img/Ecoli_20161014_OD_growthrates.png" height="250"><oq>\(\mu = k \frac{\text{ribosomes}}{\text{proteins}}\)</oq></p>

<div  style='text-align: left;line-height: 90%;'><font size=3> 
Brauer <em>et al.</em>, Mol Biol Cell 2008: <em>Coordination of growth rate, cell cycle, stress response, and metabolic activity in yeast.</em><br/>
Slavov <em>et al.</em>, Mol Biol Cell 2011: <em>Coupling among growth rate response, metabolic cycle, and cell division cycle in yeast.</em><br/>
Scott <em>et al.</em>, Science 2010: <em>Interdependence of cell growth and gene expression: origins and consequences.</em></br>
Wei&szlig;e <em>et al.</em>, PNAS 2015: <em>Mechanistic links between cellular trade-offs, gene expression, and growth.</em>
</font></div>

</div>
  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-8" style="background:;">
  <article data-timings="">
    
<div style='float:left;width:35%;'>
  <p><img src="assets/img/scott14_fig1b.png" height="220"></p>

<p><img src="assets/img/scott14_fig4.png" height="220"></p>

<p>Scott <em>et al.</em>, Mol Syst Biol 2014: <em>Emergence of robust growth laws from optimal regulation of ribosome synthesis.</em></p>

</div>
<div style='float:right;width:60%;'>
  <p><img src="assets/img/cluster_vs_rates_major.png" height="230"><br/>
Expression of large gene groups correlates with \(\mu\).</p>

<p><img src="assets/img/slavov14_selected_transcripts.png" height="230"><br/>
<oq>Even at constant \(\mu\) cells are not 
in steady-state!</oq></p>

<p>The concept of &quot;balanced growth&quot; is shakey, yet a central assumption
of many quantitative models (e.g. Rodrigo et al. 2012).</p>

</div>
  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="codefont" id="slide-9" style="background:;">
  <hgroup>
    <h3>Growth &amp; Gene Expression in <em>E. coli</em> : exponential growth</h3>
  </hgroup>
  <article data-timings="">
    
<div style='float:left;width:48%;' class='centered'>
  <p><img src="assets/img/ecoli_20141014.png" height="250"></p>

<p>\[ 
\begin{equation*} \begin{aligned}  
\frac{\text{d}X(t)}{\text{d}t} =& \mu X(t)\\ 
X(t) =& X(0)   e^{\mu  t}\\ 
\end{aligned} \end{equation*} \]</p>

</div>
<div style='float:right;width:48%;'>
  <pre><code class="r">time &lt;- seq(0,10,0.1) # hours
mu &lt;- 0.3 # specific growth rate, hour^-1
x0 &lt;- 0.01 # the inoculum: cell density, cells liter^-1
xt &lt;- x0 * exp(mu*time)
par(mai=c(.75,.75,.1,.1),mgp=c(1.5,.5,0),cex=1.2)
plot(time, xt,
     xlab=&quot;time, h&quot;,ylab=expression(X[0]*e^(mu*t)))
</code></pre>

<p><img src="assets/fig/unnamed-chunk-3-1.png" alt="plot of chunk unnamed-chunk-3"></p>

</div>
  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="codefont" id="slide-10" style="background:;">
  <hgroup>
    <h3>Growth &amp; Gene Expression in <em>E. coli</em> : growth rate</h3>
  </hgroup>
  <article data-timings="">
    
<div style='float:left;width:48%;' class='centered'>
  <p><img src="assets/img/ecoli_20141014.png" height="250"></p>

<p>\[ 
\begin{equation*} \begin{aligned}  
\frac{\text{d}X(t)}{\text{d}t} =& \mu X(t)\\ 
X(t) =& X(0)   e^{\mu  t}\\ 
\ln \frac{X(t)}{X(0)} =& \mu t
\end{aligned} \end{equation*} \]</p>

</div>
<div style='float:right;width:48%;'>
  <pre><code class="r">time &lt;- seq(0,10,0.1) # hours
mu &lt;- 0.3 # specific growth rate, hour^-1
x0 &lt;- 0.01 # the inoculum: cell density, cells liter^-1
xt &lt;- x0 * exp(mu*time)
par(mai=c(.75,.75,.1,.1),mgp=c(1.5,.5,0),cex=1.2)
plot(time, log(xt/x0),
     xlab=&quot;time, h&quot;,ylab=expression(ln(X(t)/X[0])))
</code></pre>

<p><img src="assets/fig/unnamed-chunk-4-1.png" alt="plot of chunk unnamed-chunk-4"></p>

</div>
  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="codefont" id="slide-11" style="background:;">
  <hgroup>
    <h3>Growth &amp; Gene Expression in <em>E. coli</em> : doubling time</h3>
  </hgroup>
  <article data-timings="">
    
<div style='float:left;width:48%;' class='centered'>
  <p><img src="assets/img/ecoli_20141014.png" height="250"></p>

<p>\[ 
\begin{equation*} \begin{aligned}  
\frac{\text{d}X(t)}{\text{d}t} =& \mu X(t)\\ 
X(t) =& X(0)   e^{\mu  t}\\ 
\frac{\ln 2}{\mu} = & t_D
\end{aligned} \end{equation*} \]</p>

</div>
<div style='float:right;width:48%;'>
  <pre><code class="r">time &lt;- seq(0,10,0.1) # hours
mu &lt;- 0.3 # specific growth rate, hour^-1
x0 &lt;- 0.01 # the inoculum: cell density, cells liter^-1
xt &lt;- x0 * exp(mu*time)
par(mai=c(.75,.75,.1,.1),mgp=c(1.5,.5,0),cex=1.2)
plot(time, log(xt/x0),
     xlab=&quot;time, h&quot;,ylab=expression(ln(X(t)/X[0])))
</code></pre>

<p><img src="assets/fig/unnamed-chunk-5-1.png" alt="plot of chunk unnamed-chunk-5"></p>

</div>
  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="codefont" id="slide-12" style="background:;">
  <hgroup>
    <h3>Growth &amp; Gene Expression in <em>E. coli</em> : growth rate</h3>
  </hgroup>
  <article data-timings="">
    
<div style='float:left;width:35%;'>
  <p><img src="assets/img/ecoli_20141014.png" height="250"></p>

<p>\[ 
\begin{equation*} \begin{aligned}  
\frac{\text{d}X(t)}{\text{d}t} =& \mu X(t)\\ 
X(t) =& X(0)   e^{\mu  t}\\ 
\ln(X(t)) =& \mu t + \ln(X(0))
\end{aligned} \end{equation*} \]</p>

</div>
<div style='float:right;width:60%;'>
  <pre><code class="r">par(mai=c(.75,.75,.1,.1),mgp=c(1.5,.5,0),cex=1.2)
plot(time, log(xt), xlab=&quot;time, h&quot;,ylab=expression(ln~X(t)))
x1 &lt;- .05; idx1 &lt;- which(abs(xt-x1)==min(abs(xt-x1)))
x2 &lt;- .1; idx2 &lt;- which(abs(xt-x2)==min(abs(xt-x2)))
lines(x=time[c(idx1,idx2)], y=log(xt[c(idx1,idx1)]),col=2,lwd=5)
text(time[idx2],mean(log(xt[c(idx1,idx2)])),expression(Delta~X),pos=4,col=2)
lines(x=time[c(idx2,idx2)], y=log(xt[c(idx1,idx2)]),col=2,lwd=5)
text(mean(time[c(idx1,idx2)]),log(xt[idx1]),expression(Delta~t),pos=1,col=2)
arrows(x0=0,x1=0,y0=log(xt[1]),y1=-1.4,col=2,lwd=5);
text(x=0,y=-3,expression(ln(X(0))),pos=4,col=2)
</code></pre>

<p><img src="assets/fig/unnamed-chunk-6-1.png" alt="plot of chunk unnamed-chunk-6"></p>

</div>
  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="codefont" id="slide-13" style="background:;">
  <hgroup>
    <h3>Load Your Data</h3>
  </hgroup>
  <article data-timings="">
    <pre><code class="r">library(platexpress)

setwd(&quot;~/work/hhu_2015/uebung_201612/cellgrowth_20161214/ecoli_ts_20161014&quot;)

plate &lt;-readPlateMap(&quot;20161014_platemap.csv&quot;,fsep=&quot;;&quot;,
                     fields=c(&quot;strain&quot;,&quot;IPTG&quot;,&quot;blank&quot;))
files &lt;- c(&quot;20161014_20161014 IPTG mVenus Injection  1_Absorbance.CSV&quot;,
           &quot;20161014_20161014 IPTG mVenus Injection  1_Fluorescence.CSV&quot;)
raw &lt;- readPlateData(files,type=&quot;BMG&quot;,time.conversion=1/60)
</code></pre>

<pre><code>## Parsing file 20161014_20161014 IPTG mVenus Injection  1_Absorbance.CSV 
##  found data 584 
## Parsing file 20161014_20161014 IPTG mVenus Injection  1_Fluorescence.CSV 
##  found data 485/Em520 
## Interpolating all data to a single master time.
</code></pre>

<pre><code>## Warning in listAverage(data, &quot;time&quot;): time : max. SD within timepoint is 70.5 % of median difference between time points.
</code></pre>

<p><oq> What does the warning mean?<oq/></p>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-14" style="background:;">
  <article data-timings="">
    <pre><code class="r">viewPlate(raw)
</code></pre>

<pre><code>## x-axis: Time 
## plotting 584;485/Em520
</code></pre>

<p><img src="assets/fig/unnamed-chunk-8-1.png" alt="plot of chunk unnamed-chunk-8"></p>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-15" style="background:;">
  <article data-timings="">
    <pre><code class="r">showSpectrum() #findWavelength(3)
</code></pre>

<p><img src="assets/fig/unnamed-chunk-9-1.png" alt="plot of chunk unnamed-chunk-9"></p>

<pre><code class="r">## re-name and color data
raw &lt;- prettyData(raw, dids=c(OD=&quot;584&quot;,mVenus=&quot;485/Em520&quot;),
                  colors=c(&quot;#000000&quot;,wavelength2RGB(600)))
</code></pre>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-16" style="background:;">
  <article data-timings="">
    <pre><code class="r">vp &lt;- viewPlate(raw,xlim=c(0,1800),xscale=TRUE)
</code></pre>

<pre><code>## x-axis: Time 
## plotting OD;mVenus
</code></pre>

<p><img src="assets/fig/unnamed-chunk-10-1.png" alt="plot of chunk unnamed-chunk-10"></p>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-17" style="background:;">
  <article data-timings="">
    <pre><code class="r">## GET A SINGLE DATASET
od &lt;- getData(raw,&quot;OD&quot;)
TIME &lt;- raw$Time
Xt &lt;- od[,&quot;A8&quot;]
plot(TIME, Xt)
</code></pre>

<p><img src="assets/fig/unnamed-chunk-11-1.png" alt="plot of chunk unnamed-chunk-11"></p>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-18" style="background:;">
  <article data-timings="">
    <pre><code class="r">## cut to growth range
rng &lt;- TIME&lt;1500
Xt &lt;- Xt[rng]
TIME &lt;- TIME[rng]

## look at data
par(mfcol=c(1,2),mai=c(.75,.75,.1,.1),mgp=c(1.5,.5,0),cex=1.2)
plot(TIME, Xt)
plot(TIME, log(Xt)) # log it
</code></pre>

<p><img src="assets/fig/unnamed-chunk-12-1.png" alt="plot of chunk unnamed-chunk-12"></p>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-19" style="background:;">
  <article data-timings="">
    <pre><code class="r">## cut to linear range of growth
rng &lt;- TIME&gt;600 &amp; TIME &lt; 900
xt &lt;- Xt[rng]
time &lt;- TIME[rng]

## look again at data
par(mfcol=c(1,2))
plot(time,xt)
plot(time,log(xt))
</code></pre>

<p><img src="assets/fig/unnamed-chunk-13-1.png" alt="plot of chunk unnamed-chunk-13"></p>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="codefont" id="slide-20" style="background:;">
  <article data-timings="">
    <pre><code class="r">## DO LINEAR REGRESSION
## ln(X(t)) = mu * t + ln(X(0))
lfit &lt;- lm(log(xt) ~ time)

## check quality of fit
summary(lfit)
</code></pre>

<pre><code>## 
## Call:
## lm(formula = log(xt) ~ time)
## 
## Residuals:
##       Min        1Q    Median        3Q       Max 
## -0.021694 -0.006298 -0.001451  0.006707  0.023853 
## 
## Coefficients:
##               Estimate Std. Error t value Pr(&gt;|t|)    
## (Intercept) -1.840e+00  1.122e-02  -164.0   &lt;2e-16 ***
## time         7.663e-04  1.488e-05    51.5   &lt;2e-16 ***
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1
## 
## Residual standard error: 0.01001 on 58 degrees of freedom
## Multiple R-squared:  0.9786, Adjusted R-squared:  0.9782 
## F-statistic:  2652 on 1 and 58 DF,  p-value: &lt; 2.2e-16
</code></pre>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-21" style="background:;">
  <article data-timings="">
    <pre><code class="r">## get parameters from linear regression
x0.1 &lt;- exp(coefficients(lfit)[1]) # ln(X(0))
mu.1 &lt;- coefficients(lfit)[2] # mu

## plot
par(mai=c(.75,.75,.1,.1),mgp=c(1.5,.5,0),cex=1.2)
plot(TIME,log(Xt))
lines(TIME, mu.1*TIME + log(x0.1), col=&quot;red&quot;)
</code></pre>

<p><img src="assets/fig/unnamed-chunk-15-1.png" alt="plot of chunk unnamed-chunk-15"></p>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="codefont" id="slide-22" style="background:;">
  <article data-timings="">
    <pre><code class="r">## DO NON-LINEAR REGRESSION, using
## the results of the linear fit as initial parameter guesses
dat &lt;- data.frame(time=time, xt=xt)
start &lt;- list(mu=mu.1,x0=x0.1)
nlfit &lt;- nls(xt ~ x0*exp(mu*time),data=dat,start=start)

## check quality of fit
summary(nlfit)
</code></pre>

<pre><code>## 
## Formula: xt ~ x0 * exp(mu * time)
## 
## Parameters:
##     Estimate Std. Error t value Pr(&gt;|t|)    
## mu 7.660e-04  1.474e-05   51.99   &lt;2e-16 ***
## x0 1.588e-01  1.792e-03   88.63   &lt;2e-16 ***
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1
## 
## Residual standard error: 0.002792 on 58 degrees of freedom
## 
## Number of iterations to convergence: 1 
## Achieved convergence tolerance: 2.422e-06
</code></pre>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="codefont" id="slide-23" style="background:;">
  <article data-timings="">
    <pre><code class="r">## get parameters &amp; plot results
mu.2 &lt;- coefficients(nlfit)[1]
x0.2 &lt;- coefficients(nlfit)[2]
par(mfcol=c(1,2),mai=c(.75,.75,.1,.1),mgp=c(1.5,.5,0),cex=1.2)
plot(TIME,Xt,ylim=c(0.2,.4))
lines(TIME, x0.2 * exp(TIME*mu.2),col=&quot;green&quot;, lty=2,lwd=5)
lines(TIME, x0.1 * exp(TIME*mu.1),col=&quot;red&quot;,lwd=2)
legend(&quot;bottomright&quot;,legend=c(&quot;data&quot;,&quot;lin.reg.&quot;,&quot;non-lin.&quot;),
       col=c(1,2,3),pch=c(1,NA,NA),lty=c(NA,1,2),lwd=3)
plot(TIME, x0.2 * exp(TIME*mu.2),col=&quot;green&quot;, lty=2,lwd=5)
points(TIME,Xt)
lines(TIME, x0.1 * exp(TIME*mu.1),col=&quot;red&quot;,lwd=2)
</code></pre>

<p><img src="assets/fig/unnamed-chunk-17-1.png" alt="plot of chunk unnamed-chunk-17"></p>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-24" style="background:;">
  <hgroup>
    <h3>Fitting Growth Models</h3>
  </hgroup>
  <article data-timings="">
    
<div style='float:left;width:48%;' class='centered'>
  <p><img src="assets/img/Ecoli_20161014_OD_grofit_A8.png" height="300"></p>

<ul>
<li>Initial Cell Density: \(X(0) \)</li>
<li>Lag Phase: \(\lambda \)</li>
<li>Exponential Phase: growth rate \(\mu \)</li>
<li>Stationary Phase: <em>capacity</em> \(A \) </li>
</ul>

</div>
<div style='float:right;width:48%;'>
  <ul>
<li>Logistic Equation:<br/>
\(X(t) = \frac{A}{1+e^{\frac{4 \mu}{A}(\lambda - t)+1)}} \)</li>
<li>Gompertz:<br/>
\(X(t) = A e^{-e^{\frac{\mu e}{A}(\lambda -t)+1}} \)</li>
<li>Modified Gompertz:<br/>
\(X(t) = A e^{-e^{\frac{\mu e}{A}(\lambda -t)+1}} + A e^{\alpha(t-t_{shift})} \)</li>
<li>Richard&#39;s generalized logistic model:<br/> \(  X(t) = A (1 + \nu e^{1+ \nu + \frac{\mu}{A}(1+\nu )^{1+\frac{1}{\nu}}(\lambda -t)})^{-\frac{1}{\nu}} \)</li>
</ul>

<p>as implemented in R package <code>grofit</code></p>

<p><oq> Try these equations with <code>nls</code>.<p/></p>

</div>
  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-25" style="background:;">
  <hgroup>
    <h3>Prepare Data: blanks, cuts, etc.</h3>
  </hgroup>
  <article data-timings="">
    <pre><code class="r">raw2 &lt;- cutData(raw, rng=c(0,1550), mid=&quot;Time&quot;)
raw3 &lt;- correctBlanks(raw2, plate,dids=&quot;OD&quot;,max.mid=1500) 
</code></pre>

<pre><code>## blanking OD 
## blanking TRUE : 84 wells, using 12 blank wells
## OD 
##  time bin: 1 1 - 309 skipping 10 bins at 1500 
##  blank: 0.2435858298247
</code></pre>

<pre><code class="r">raw4 &lt;- correctBlanks(raw3,plate,dids=&quot;mVenus&quot;,by=c(&quot;strain&quot;,&quot;IPTG&quot;),
                      mbins=length(raw$Time)/5,verb=FALSE)
data &lt;- adjustBase(raw4, base=0, add.fraction=.001, 
                   wells=unlist(groups),xlim=c(1,which(raw$Time&gt;1500)[1]),
                   each=TRUE, verb=FALSE) # set verb to TRUE
</code></pre>

<p><oq> What happens in <code>correctBlanks</code> and <code>cutData</code>? <br/>
What does <code>adjustBase</code> do, and when could we need it?<p/></p>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-26" style="background:;">
  <hgroup>
    <h3>Get Replicate Groups</h3>
  </hgroup>
  <article data-timings="">
    <pre><code class="r">groups &lt;- getGroups(plate, c(&quot;strain&quot;), verb=FALSE) # SET TO TRUE!
groups2 &lt;- getGroups(plate, c(&quot;strain&quot;,&quot;IPTG&quot;), verb=FALSE) 
viewGroups(data, groups=groups, groups2=groups2, verb=FALSE)
</code></pre>

<p><img src="assets/fig/unnamed-chunk-19-1.png" alt="plot of chunk unnamed-chunk-19"></p>

<p><oq> What are the lines and areas?<br/>
What is the structure of the <code>groups</code> item? 
Can you make your own groupings?<br/>
Try different groupings and parameters to <code>viewGroups</code>.</oq></p>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-27" style="background:;">
  <hgroup>
    <h3>Use Package <code>grofit</code> to Fit Growth Data</h3>
  </hgroup>
  <article data-timings="">
    <pre><code class="r">raw2 &lt;- cutData(raw,rng=c(200,1500))
grodat &lt;- data2grofit(raw2, did=&quot;OD&quot;, plate=plate, 
                      wells=groups[[&quot;pUC18mV&quot;]])
library(grofit)
fitparams &lt;- grofit.2.control(interactive=FALSE, plot=TRUE) # SET ALL TO TRUE!!
pdf(&quot;growthrates.pdf&quot;)
fits &lt;- gcFit.2(time=grodat$time, data=grodat$data, control=fitparams)
</code></pre>

<pre><code>## 
## 
## = 1. growth curve =================================
## ----------------------------------------------------
## --&gt; Try to fit model logistic
</code></pre>

<pre><code>## ....... OK
</code></pre>

<pre><code>## --&gt; Try to fit model richards
</code></pre>

<pre><code>## ....... ERROR in nls(). For further information see help(gcFitModel)
</code></pre>

<pre><code>## --&gt; Try to fit model gompertz
</code></pre>

<pre><code>## ....... OK
</code></pre>

<pre><code>## --&gt; Try to fit model gompertz.exp
</code></pre>

<pre><code>## ... ERROR in nls(). For further information see help(gcFitModel)
</code></pre>

<pre><code>## 
## 
## = 2. growth curve =================================
## ----------------------------------------------------
## --&gt; Try to fit model logistic
</code></pre>

<pre><code>## ....... OK
</code></pre>

<pre><code>## --&gt; Try to fit model richards
</code></pre>

<pre><code>## ....... OK
</code></pre>

<pre><code>## --&gt; Try to fit model gompertz
</code></pre>

<pre><code>## ....... OK
</code></pre>

<pre><code>## --&gt; Try to fit model gompertz.exp
</code></pre>

<pre><code>## ... ERROR in nls(). For further information see help(gcFitModel)
</code></pre>

<pre><code>## 
## 
## = 3. growth curve =================================
## ----------------------------------------------------
## --&gt; Try to fit model logistic
</code></pre>

<pre><code>## ....... OK
</code></pre>

<pre><code>## --&gt; Try to fit model richards
</code></pre>

<pre><code>## ....... OK
</code></pre>

<pre><code>## --&gt; Try to fit model gompertz
</code></pre>

<pre><code>## ....... OK
</code></pre>

<pre><code>## --&gt; Try to fit model gompertz.exp
</code></pre>

<pre><code>## ... ERROR in nls(). For further information see help(gcFitModel)
</code></pre>

<pre><code>## 
## 
## = 4. growth curve =================================
## ----------------------------------------------------
## --&gt; Try to fit model logistic
</code></pre>

<pre><code>## ....... OK
</code></pre>

<pre><code>## --&gt; Try to fit model richards
</code></pre>

<pre><code>## ....... ERROR in nls(). For further information see help(gcFitModel)
</code></pre>

<pre><code>## --&gt; Try to fit model gompertz
</code></pre>

<pre><code>## ....... OK
</code></pre>

<pre><code>## --&gt; Try to fit model gompertz.exp
</code></pre>

<pre><code>## ... ERROR in nls(). For further information see help(gcFitModel)
</code></pre>

<pre><code>## 
## 
## = 5. growth curve =================================
## ----------------------------------------------------
## --&gt; Try to fit model logistic
</code></pre>

<pre><code>## ....... OK
</code></pre>

<pre><code>## --&gt; Try to fit model richards
</code></pre>

<pre><code>## ....... ERROR in nls(). For further information see help(gcFitModel)
</code></pre>

<pre><code>## --&gt; Try to fit model gompertz
</code></pre>

<pre><code>## ....... OK
</code></pre>

<pre><code>## --&gt; Try to fit model gompertz.exp
</code></pre>

<pre><code>## ... ERROR in nls(). For further information see help(gcFitModel)
</code></pre>

<pre><code>## 
## 
## = 6. growth curve =================================
## ----------------------------------------------------
## --&gt; Try to fit model logistic
</code></pre>

<pre><code>## ....... OK
</code></pre>

<pre><code>## --&gt; Try to fit model richards
</code></pre>

<pre><code>## ....... ERROR in nls(). For further information see help(gcFitModel)
</code></pre>

<pre><code>## --&gt; Try to fit model gompertz
</code></pre>

<pre><code>## ....... OK
</code></pre>

<pre><code>## --&gt; Try to fit model gompertz.exp
</code></pre>

<pre><code>## ... ERROR in nls(). For further information see help(gcFitModel)
</code></pre>

<pre><code>## 
## 
## = 7. growth curve =================================
## ----------------------------------------------------
## --&gt; Try to fit model logistic
</code></pre>

<pre><code>## ....... OK
</code></pre>

<pre><code>## --&gt; Try to fit model richards
</code></pre>

<pre><code>## ....... ERROR in nls(). For further information see help(gcFitModel)
</code></pre>

<pre><code>## --&gt; Try to fit model gompertz
</code></pre>

<pre><code>## ....... OK
</code></pre>

<pre><code>## --&gt; Try to fit model gompertz.exp
</code></pre>

<pre><code>## ... ERROR in nls(). For further information see help(gcFitModel)
</code></pre>

<pre><code>## 
## 
## = 8. growth curve =================================
## ----------------------------------------------------
## --&gt; Try to fit model logistic
</code></pre>

<pre><code>## ....... OK
</code></pre>

<pre><code>## --&gt; Try to fit model richards
</code></pre>

<pre><code>## ....... OK
</code></pre>

<pre><code>## --&gt; Try to fit model gompertz
</code></pre>

<pre><code>## ....... OK
</code></pre>

<pre><code>## --&gt; Try to fit model gompertz.exp
</code></pre>

<pre><code>## ... ERROR in nls(). For further information see help(gcFitModel)
</code></pre>

<pre><code>## 
## 
## = 9. growth curve =================================
## ----------------------------------------------------
## --&gt; Try to fit model logistic
</code></pre>

<pre><code>## ....... OK
</code></pre>

<pre><code>## --&gt; Try to fit model richards
</code></pre>

<pre><code>## ....... ERROR in nls(). For further information see help(gcFitModel)
</code></pre>

<pre><code>## --&gt; Try to fit model gompertz
</code></pre>

<pre><code>## ....... ERROR in nls(). For further information see help(gcFitModel)
</code></pre>

<pre><code>## --&gt; Try to fit model gompertz.exp
</code></pre>

<pre><code>## ... ERROR in nls(). For further information see help(gcFitModel)
</code></pre>

<pre><code>## 
## 
## = 10. growth curve =================================
## ----------------------------------------------------
## --&gt; Try to fit model logistic
</code></pre>

<pre><code>## ....... OK
</code></pre>

<pre><code>## --&gt; Try to fit model richards
</code></pre>

<pre><code>## ....... ERROR in nls(). For further information see help(gcFitModel)
</code></pre>

<pre><code>## --&gt; Try to fit model gompertz
</code></pre>

<pre><code>## ....... ERROR in nls(). For further information see help(gcFitModel)
</code></pre>

<pre><code>## --&gt; Try to fit model gompertz.exp
</code></pre>

<pre><code>## ... ERROR in nls(). For further information see help(gcFitModel)
</code></pre>

<pre><code>## 
## 
## = 11. growth curve =================================
## ----------------------------------------------------
## --&gt; Try to fit model logistic
</code></pre>

<pre><code>## ....... OK
</code></pre>

<pre><code>## --&gt; Try to fit model richards
</code></pre>

<pre><code>## ....... ERROR in nls(). For further information see help(gcFitModel)
</code></pre>

<pre><code>## --&gt; Try to fit model gompertz
</code></pre>

<pre><code>## ....... ERROR in nls(). For further information see help(gcFitModel)
</code></pre>

<pre><code>## --&gt; Try to fit model gompertz.exp
</code></pre>

<pre><code>## ... ERROR in nls(). For further information see help(gcFitModel)
</code></pre>

<pre><code>## 
## 
## = 12. growth curve =================================
## ----------------------------------------------------
## --&gt; Try to fit model logistic
</code></pre>

<pre><code>## ....... ERROR in nls(). For further information see help(gcFitModel)
</code></pre>

<pre><code>## --&gt; Try to fit model richards
</code></pre>

<pre><code>## ....... ERROR in nls(). For further information see help(gcFitModel)
</code></pre>

<pre><code>## --&gt; Try to fit model gompertz
</code></pre>

<pre><code>## ....... ERROR in nls(). For further information see help(gcFitModel)
</code></pre>

<pre><code>## --&gt; Try to fit model gompertz.exp
</code></pre>

<pre><code>## ... ERROR in nls(). For further information see help(gcFitModel)
</code></pre>

<pre><code>## Warning in grofit::gcFitModel(acttime, actwell, gcID, control): gcFitModel:
## Unable to fit this curve parametrically!
</code></pre>

<pre><code>## 
## 
## = 13. growth curve =================================
## ----------------------------------------------------
## --&gt; Try to fit model logistic
</code></pre>

<pre><code>## ....... OK
</code></pre>

<pre><code>## --&gt; Try to fit model richards
</code></pre>

<pre><code>## ....... ERROR in nls(). For further information see help(gcFitModel)
</code></pre>

<pre><code>## --&gt; Try to fit model gompertz
</code></pre>

<pre><code>## ....... OK
</code></pre>

<pre><code>## --&gt; Try to fit model gompertz.exp
</code></pre>

<pre><code>## ... ERROR in nls(). For further information see help(gcFitModel)
</code></pre>

<pre><code>## 
## 
## = 14. growth curve =================================
## ----------------------------------------------------
## --&gt; Try to fit model logistic
</code></pre>

<pre><code>## ....... OK
</code></pre>

<pre><code>## --&gt; Try to fit model richards
</code></pre>

<pre><code>## ....... ERROR in nls(). For further information see help(gcFitModel)
</code></pre>

<pre><code>## --&gt; Try to fit model gompertz
</code></pre>

<pre><code>## ....... OK
</code></pre>

<pre><code>## --&gt; Try to fit model gompertz.exp
</code></pre>

<pre><code>## ... ERROR in nls(). For further information see help(gcFitModel)
</code></pre>

<pre><code>## 
## 
## = 15. growth curve =================================
## ----------------------------------------------------
## --&gt; Try to fit model logistic
</code></pre>

<pre><code>## ....... OK
</code></pre>

<pre><code>## --&gt; Try to fit model richards
</code></pre>

<pre><code>## ....... ERROR in nls(). For further information see help(gcFitModel)
</code></pre>

<pre><code>## --&gt; Try to fit model gompertz
</code></pre>

<pre><code>## ....... OK
</code></pre>

<pre><code>## --&gt; Try to fit model gompertz.exp
</code></pre>

<pre><code>## ... ERROR in nls(). For further information see help(gcFitModel)
</code></pre>

<pre><code>## 
## 
## = 16. growth curve =================================
## ----------------------------------------------------
## --&gt; Try to fit model logistic
</code></pre>

<pre><code>## ....... OK
</code></pre>

<pre><code>## --&gt; Try to fit model richards
</code></pre>

<pre><code>## ....... ERROR in nls(). For further information see help(gcFitModel)
</code></pre>

<pre><code>## --&gt; Try to fit model gompertz
</code></pre>

<pre><code>## ....... OK
</code></pre>

<pre><code>## --&gt; Try to fit model gompertz.exp
</code></pre>

<pre><code>## ... ERROR in nls(). For further information see help(gcFitModel)
</code></pre>

<pre><code>## 
## 
## = 17. growth curve =================================
## ----------------------------------------------------
## --&gt; Try to fit model logistic
</code></pre>

<pre><code>## ....... OK
</code></pre>

<pre><code>## --&gt; Try to fit model richards
</code></pre>

<pre><code>## ....... ERROR in nls(). For further information see help(gcFitModel)
</code></pre>

<pre><code>## --&gt; Try to fit model gompertz
</code></pre>

<pre><code>## ....... OK
</code></pre>

<pre><code>## --&gt; Try to fit model gompertz.exp
</code></pre>

<pre><code>## ... ERROR in nls(). For further information see help(gcFitModel)
</code></pre>

<pre><code>## 
## 
## = 18. growth curve =================================
## ----------------------------------------------------
## --&gt; Try to fit model logistic
</code></pre>

<pre><code>## ....... OK
</code></pre>

<pre><code>## --&gt; Try to fit model richards
</code></pre>

<pre><code>## ....... ERROR in nls(). For further information see help(gcFitModel)
</code></pre>

<pre><code>## --&gt; Try to fit model gompertz
</code></pre>

<pre><code>## ....... OK
</code></pre>

<pre><code>## --&gt; Try to fit model gompertz.exp
</code></pre>

<pre><code>## ... ERROR in nls(). For further information see help(gcFitModel)
</code></pre>

<pre><code>## 
## 
## = 19. growth curve =================================
## ----------------------------------------------------
## --&gt; Try to fit model logistic
</code></pre>

<pre><code>## ....... OK
</code></pre>

<pre><code>## --&gt; Try to fit model richards
</code></pre>

<pre><code>## ....... ERROR in nls(). For further information see help(gcFitModel)
</code></pre>

<pre><code>## --&gt; Try to fit model gompertz
</code></pre>

<pre><code>## ....... OK
</code></pre>

<pre><code>## --&gt; Try to fit model gompertz.exp
</code></pre>

<pre><code>## ... ERROR in nls(). For further information see help(gcFitModel)
</code></pre>

<pre><code>## 
## 
## = 20. growth curve =================================
## ----------------------------------------------------
## --&gt; Try to fit model logistic
</code></pre>

<pre><code>## ....... OK
</code></pre>

<pre><code>## --&gt; Try to fit model richards
</code></pre>

<pre><code>## ....... ERROR in nls(). For further information see help(gcFitModel)
</code></pre>

<pre><code>## --&gt; Try to fit model gompertz
</code></pre>

<pre><code>## ....... OK
</code></pre>

<pre><code>## --&gt; Try to fit model gompertz.exp
</code></pre>

<pre><code>## ... ERROR in nls(). For further information see help(gcFitModel)
</code></pre>

<pre><code>## 
## 
## = 21. growth curve =================================
## ----------------------------------------------------
## --&gt; Try to fit model logistic
</code></pre>

<pre><code>## ....... OK
</code></pre>

<pre><code>## --&gt; Try to fit model richards
</code></pre>

<pre><code>## ....... ERROR in nls(). For further information see help(gcFitModel)
</code></pre>

<pre><code>## --&gt; Try to fit model gompertz
</code></pre>

<pre><code>## ....... OK
</code></pre>

<pre><code>## --&gt; Try to fit model gompertz.exp
</code></pre>

<pre><code>## ... ERROR in nls(). For further information see help(gcFitModel)
</code></pre>

<pre><code>## 
## 
## = 22. growth curve =================================
## ----------------------------------------------------
## --&gt; Try to fit model logistic
</code></pre>

<pre><code>## ....... OK
</code></pre>

<pre><code>## --&gt; Try to fit model richards
</code></pre>

<pre><code>## ....... ERROR in nls(). For further information see help(gcFitModel)
</code></pre>

<pre><code>## --&gt; Try to fit model gompertz
</code></pre>

<pre><code>## ....... OK
</code></pre>

<pre><code>## --&gt; Try to fit model gompertz.exp
</code></pre>

<pre><code>## ... ERROR in nls(). For further information see help(gcFitModel)
</code></pre>

<pre><code>## 
## 
## = 23. growth curve =================================
## ----------------------------------------------------
## --&gt; Try to fit model logistic
</code></pre>

<pre><code>## ....... OK
</code></pre>

<pre><code>## --&gt; Try to fit model richards
</code></pre>

<pre><code>## ....... ERROR in nls(). For further information see help(gcFitModel)
</code></pre>

<pre><code>## --&gt; Try to fit model gompertz
</code></pre>

<pre><code>## ....... OK
</code></pre>

<pre><code>## --&gt; Try to fit model gompertz.exp
</code></pre>

<pre><code>## ... ERROR in nls(). For further information see help(gcFitModel)
</code></pre>

<pre><code>## 
## 
## = 24. growth curve =================================
## ----------------------------------------------------
## --&gt; Try to fit model logistic
</code></pre>

<pre><code>## ....... OK
</code></pre>

<pre><code>## --&gt; Try to fit model richards
</code></pre>

<pre><code>## ....... ERROR in nls(). For further information see help(gcFitModel)
</code></pre>

<pre><code>## --&gt; Try to fit model gompertz
</code></pre>

<pre><code>## ....... OK
</code></pre>

<pre><code>## --&gt; Try to fit model gompertz.exp
</code></pre>

<pre><code>## ... ERROR in nls(). For further information see help(gcFitModel)
</code></pre>

<pre><code>## 
## 
## = 25. growth curve =================================
## ----------------------------------------------------
## --&gt; Try to fit model logistic
</code></pre>

<pre><code>## ....... OK
</code></pre>

<pre><code>## --&gt; Try to fit model richards
</code></pre>

<pre><code>## ....... ERROR in nls(). For further information see help(gcFitModel)
</code></pre>

<pre><code>## --&gt; Try to fit model gompertz
</code></pre>

<pre><code>## ....... OK
</code></pre>

<pre><code>## --&gt; Try to fit model gompertz.exp
</code></pre>

<pre><code>## ... ERROR in nls(). For further information see help(gcFitModel)
</code></pre>

<pre><code>## 
## 
## = 26. growth curve =================================
## ----------------------------------------------------
## --&gt; Try to fit model logistic
</code></pre>

<pre><code>## ....... OK
</code></pre>

<pre><code>## --&gt; Try to fit model richards
</code></pre>

<pre><code>## ....... ERROR in nls(). For further information see help(gcFitModel)
</code></pre>

<pre><code>## --&gt; Try to fit model gompertz
</code></pre>

<pre><code>## ....... OK
</code></pre>

<pre><code>## --&gt; Try to fit model gompertz.exp
</code></pre>

<pre><code>## ... ERROR in nls(). For further information see help(gcFitModel)
</code></pre>

<pre><code>## 
## 
## = 27. growth curve =================================
## ----------------------------------------------------
## --&gt; Try to fit model logistic
</code></pre>

<pre><code>## ....... OK
</code></pre>

<pre><code>## --&gt; Try to fit model richards
</code></pre>

<pre><code>## ....... ERROR in nls(). For further information see help(gcFitModel)
</code></pre>

<pre><code>## --&gt; Try to fit model gompertz
</code></pre>

<pre><code>## ....... OK
</code></pre>

<pre><code>## --&gt; Try to fit model gompertz.exp
</code></pre>

<pre><code>## ... ERROR in nls(). For further information see help(gcFitModel)
</code></pre>

<pre><code>## 
## 
## = 28. growth curve =================================
## ----------------------------------------------------
## --&gt; Try to fit model logistic
</code></pre>

<pre><code>## ....... OK
</code></pre>

<pre><code>## --&gt; Try to fit model richards
</code></pre>

<pre><code>## ....... ERROR in nls(). For further information see help(gcFitModel)
</code></pre>

<pre><code>## --&gt; Try to fit model gompertz
</code></pre>

<pre><code>## ....... OK
</code></pre>

<pre><code>## --&gt; Try to fit model gompertz.exp
</code></pre>

<pre><code>## ... ERROR in nls(). For further information see help(gcFitModel)
</code></pre>

<pre><code>## 
## 
## = 29. growth curve =================================
## ----------------------------------------------------
## --&gt; Try to fit model logistic
</code></pre>

<pre><code>## ....... OK
</code></pre>

<pre><code>## --&gt; Try to fit model richards
</code></pre>

<pre><code>## ....... ERROR in nls(). For further information see help(gcFitModel)
</code></pre>

<pre><code>## --&gt; Try to fit model gompertz
</code></pre>

<pre><code>## ....... OK
</code></pre>

<pre><code>## --&gt; Try to fit model gompertz.exp
</code></pre>

<pre><code>## ... ERROR in nls(). For further information see help(gcFitModel)
</code></pre>

<pre><code>## 
## 
## = 30. growth curve =================================
## ----------------------------------------------------
## --&gt; Try to fit model logistic
</code></pre>

<pre><code>## ....... OK
</code></pre>

<pre><code>## --&gt; Try to fit model richards
</code></pre>

<pre><code>## ....... ERROR in nls(). For further information see help(gcFitModel)
</code></pre>

<pre><code>## --&gt; Try to fit model gompertz
</code></pre>

<pre><code>## ....... OK
</code></pre>

<pre><code>## --&gt; Try to fit model gompertz.exp
</code></pre>

<pre><code>## ... ERROR in nls(). For further information see help(gcFitModel)
</code></pre>

<pre><code>## 
## 
## = 31. growth curve =================================
## ----------------------------------------------------
## --&gt; Try to fit model logistic
</code></pre>

<pre><code>## ....... OK
</code></pre>

<pre><code>## --&gt; Try to fit model richards
</code></pre>

<pre><code>## ....... ERROR in nls(). For further information see help(gcFitModel)
</code></pre>

<pre><code>## --&gt; Try to fit model gompertz
</code></pre>

<pre><code>## ....... OK
</code></pre>

<pre><code>## --&gt; Try to fit model gompertz.exp
</code></pre>

<pre><code>## ... ERROR in nls(). For further information see help(gcFitModel)
</code></pre>

<pre><code>## 
## 
## = 32. growth curve =================================
## ----------------------------------------------------
## --&gt; Try to fit model logistic
</code></pre>

<pre><code>## ....... OK
</code></pre>

<pre><code>## --&gt; Try to fit model richards
</code></pre>

<pre><code>## ....... ERROR in nls(). For further information see help(gcFitModel)
</code></pre>

<pre><code>## --&gt; Try to fit model gompertz
</code></pre>

<pre><code>## ....... OK
</code></pre>

<pre><code>## --&gt; Try to fit model gompertz.exp
</code></pre>

<pre><code>## ... ERROR in nls(). For further information see help(gcFitModel)
</code></pre>

<pre><code>## 
## 
## = 33. growth curve =================================
## ----------------------------------------------------
## --&gt; Try to fit model logistic
</code></pre>

<pre><code>## ....... OK
</code></pre>

<pre><code>## --&gt; Try to fit model richards
</code></pre>

<pre><code>## ....... ERROR in nls(). For further information see help(gcFitModel)
</code></pre>

<pre><code>## --&gt; Try to fit model gompertz
</code></pre>

<pre><code>## ....... OK
</code></pre>

<pre><code>## --&gt; Try to fit model gompertz.exp
</code></pre>

<pre><code>## ... ERROR in nls(). For further information see help(gcFitModel)
</code></pre>

<pre><code>## 
## 
## = 34. growth curve =================================
## ----------------------------------------------------
## --&gt; Try to fit model logistic
</code></pre>

<pre><code>## ....... OK
</code></pre>

<pre><code>## --&gt; Try to fit model richards
</code></pre>

<pre><code>## ....... ERROR in nls(). For further information see help(gcFitModel)
</code></pre>

<pre><code>## --&gt; Try to fit model gompertz
</code></pre>

<pre><code>## ....... OK
</code></pre>

<pre><code>## --&gt; Try to fit model gompertz.exp
</code></pre>

<pre><code>## ... ERROR in nls(). For further information see help(gcFitModel)
</code></pre>

<pre><code>## 
## 
## = 35. growth curve =================================
## ----------------------------------------------------
## --&gt; Try to fit model logistic
</code></pre>

<pre><code>## ....... OK
</code></pre>

<pre><code>## --&gt; Try to fit model richards
</code></pre>

<pre><code>## ....... ERROR in nls(). For further information see help(gcFitModel)
</code></pre>

<pre><code>## --&gt; Try to fit model gompertz
</code></pre>

<pre><code>## ....... OK
</code></pre>

<pre><code>## --&gt; Try to fit model gompertz.exp
</code></pre>

<pre><code>## ... ERROR in nls(). For further information see help(gcFitModel)
</code></pre>

<pre><code>## 
## 
## = 36. growth curve =================================
## ----------------------------------------------------
## --&gt; Try to fit model logistic
</code></pre>

<pre><code>## ....... OK
</code></pre>

<pre><code>## --&gt; Try to fit model richards
</code></pre>

<pre><code>## ....... ERROR in nls(). For further information see help(gcFitModel)
</code></pre>

<pre><code>## --&gt; Try to fit model gompertz
</code></pre>

<pre><code>## ....... OK
</code></pre>

<pre><code>## --&gt; Try to fit model gompertz.exp
</code></pre>

<pre><code>## ... ERROR in nls(). For further information see help(gcFitModel)
</code></pre>

<pre><code>## 
## 
## = 37. growth curve =================================
## ----------------------------------------------------
## --&gt; Try to fit model logistic
</code></pre>

<pre><code>## ....... OK
</code></pre>

<pre><code>## --&gt; Try to fit model richards
</code></pre>

<pre><code>## ....... ERROR in nls(). For further information see help(gcFitModel)
</code></pre>

<pre><code>## --&gt; Try to fit model gompertz
</code></pre>

<pre><code>## ....... OK
</code></pre>

<pre><code>## --&gt; Try to fit model gompertz.exp
</code></pre>

<pre><code>## ... ERROR in nls(). For further information see help(gcFitModel)
</code></pre>

<pre><code>## 
## 
## = 38. growth curve =================================
## ----------------------------------------------------
## --&gt; Try to fit model logistic
</code></pre>

<pre><code>## ....... OK
</code></pre>

<pre><code>## --&gt; Try to fit model richards
</code></pre>

<pre><code>## ....... ERROR in nls(). For further information see help(gcFitModel)
</code></pre>

<pre><code>## --&gt; Try to fit model gompertz
</code></pre>

<pre><code>## ....... OK
</code></pre>

<pre><code>## --&gt; Try to fit model gompertz.exp
</code></pre>

<pre><code>## ... ERROR in nls(). For further information see help(gcFitModel)
</code></pre>

<pre><code>## 
## 
## = 39. growth curve =================================
## ----------------------------------------------------
## --&gt; Try to fit model logistic
</code></pre>

<pre><code>## ....... OK
</code></pre>

<pre><code>## --&gt; Try to fit model richards
</code></pre>

<pre><code>## ....... ERROR in nls(). For further information see help(gcFitModel)
</code></pre>

<pre><code>## --&gt; Try to fit model gompertz
</code></pre>

<pre><code>## ....... OK
</code></pre>

<pre><code>## --&gt; Try to fit model gompertz.exp
</code></pre>

<pre><code>## ... ERROR in nls(). For further information see help(gcFitModel)
</code></pre>

<pre><code>## 
## 
## = 40. growth curve =================================
## ----------------------------------------------------
## --&gt; Try to fit model logistic
</code></pre>

<pre><code>## ....... OK
</code></pre>

<pre><code>## --&gt; Try to fit model richards
</code></pre>

<pre><code>## ....... ERROR in nls(). For further information see help(gcFitModel)
</code></pre>

<pre><code>## --&gt; Try to fit model gompertz
</code></pre>

<pre><code>## ....... OK
</code></pre>

<pre><code>## --&gt; Try to fit model gompertz.exp
</code></pre>

<pre><code>## ... ERROR in nls(). For further information see help(gcFitModel)
</code></pre>

<pre><code>## 
## 
## = 41. growth curve =================================
## ----------------------------------------------------
## --&gt; Try to fit model logistic
</code></pre>

<pre><code>## ....... OK
</code></pre>

<pre><code>## --&gt; Try to fit model richards
</code></pre>

<pre><code>## ....... ERROR in nls(). For further information see help(gcFitModel)
</code></pre>

<pre><code>## --&gt; Try to fit model gompertz
</code></pre>

<pre><code>## ....... OK
</code></pre>

<pre><code>## --&gt; Try to fit model gompertz.exp
</code></pre>

<pre><code>## ... ERROR in nls(). For further information see help(gcFitModel)
</code></pre>

<pre><code>## 
## 
## = 42. growth curve =================================
## ----------------------------------------------------
## --&gt; Try to fit model logistic
</code></pre>

<pre><code>## ....... OK
</code></pre>

<pre><code>## --&gt; Try to fit model richards
</code></pre>

<pre><code>## ....... ERROR in nls(). For further information see help(gcFitModel)
</code></pre>

<pre><code>## --&gt; Try to fit model gompertz
</code></pre>

<pre><code>## ....... OK
</code></pre>

<pre><code>## --&gt; Try to fit model gompertz.exp
</code></pre>

<pre><code>## ... ERROR in nls(). For further information see help(gcFitModel)
</code></pre>

<pre><code>## 
## 
## = 43. growth curve =================================
## ----------------------------------------------------
## --&gt; Try to fit model logistic
</code></pre>

<pre><code>## ....... OK
</code></pre>

<pre><code>## --&gt; Try to fit model richards
</code></pre>

<pre><code>## ....... ERROR in nls(). For further information see help(gcFitModel)
</code></pre>

<pre><code>## --&gt; Try to fit model gompertz
</code></pre>

<pre><code>## ....... OK
</code></pre>

<pre><code>## --&gt; Try to fit model gompertz.exp
</code></pre>

<pre><code>## ... ERROR in nls(). For further information see help(gcFitModel)
</code></pre>

<pre><code>## 
## 
## = 44. growth curve =================================
## ----------------------------------------------------
## --&gt; Try to fit model logistic
</code></pre>

<pre><code>## ....... OK
</code></pre>

<pre><code>## --&gt; Try to fit model richards
</code></pre>

<pre><code>## ....... ERROR in nls(). For further information see help(gcFitModel)
</code></pre>

<pre><code>## --&gt; Try to fit model gompertz
</code></pre>

<pre><code>## ....... OK
</code></pre>

<pre><code>## --&gt; Try to fit model gompertz.exp
</code></pre>

<pre><code>## ... ERROR in nls(). For further information see help(gcFitModel)
</code></pre>

<pre><code>## 
## 
## = 45. growth curve =================================
## ----------------------------------------------------
## --&gt; Try to fit model logistic
</code></pre>

<pre><code>## ....... OK
</code></pre>

<pre><code>## --&gt; Try to fit model richards
</code></pre>

<pre><code>## ....... ERROR in nls(). For further information see help(gcFitModel)
</code></pre>

<pre><code>## --&gt; Try to fit model gompertz
</code></pre>

<pre><code>## ....... OK
</code></pre>

<pre><code>## --&gt; Try to fit model gompertz.exp
</code></pre>

<pre><code>## ... ERROR in nls(). For further information see help(gcFitModel)
</code></pre>

<pre><code>## 
## 
## = 46. growth curve =================================
## ----------------------------------------------------
## --&gt; Try to fit model logistic
</code></pre>

<pre><code>## ....... OK
</code></pre>

<pre><code>## --&gt; Try to fit model richards
</code></pre>

<pre><code>## ....... ERROR in nls(). For further information see help(gcFitModel)
</code></pre>

<pre><code>## --&gt; Try to fit model gompertz
</code></pre>

<pre><code>## ....... OK
</code></pre>

<pre><code>## --&gt; Try to fit model gompertz.exp
</code></pre>

<pre><code>## ... ERROR in nls(). For further information see help(gcFitModel)
</code></pre>

<pre><code>## 
## 
## = 47. growth curve =================================
## ----------------------------------------------------
## --&gt; Try to fit model logistic
</code></pre>

<pre><code>## ....... OK
</code></pre>

<pre><code>## --&gt; Try to fit model richards
</code></pre>

<pre><code>## ....... ERROR in nls(). For further information see help(gcFitModel)
</code></pre>

<pre><code>## --&gt; Try to fit model gompertz
</code></pre>

<pre><code>## ....... OK
</code></pre>

<pre><code>## --&gt; Try to fit model gompertz.exp
</code></pre>

<pre><code>## ... ERROR in nls(). For further information see help(gcFitModel)
</code></pre>

<pre><code>## 
## 
## = 48. growth curve =================================
## ----------------------------------------------------
## --&gt; Try to fit model logistic
</code></pre>

<pre><code>## ....... OK
</code></pre>

<pre><code>## --&gt; Try to fit model richards
</code></pre>

<pre><code>## ....... ERROR in nls(). For further information see help(gcFitModel)
</code></pre>

<pre><code>## --&gt; Try to fit model gompertz
</code></pre>

<pre><code>## ....... OK
</code></pre>

<pre><code>## --&gt; Try to fit model gompertz.exp
</code></pre>

<pre><code>## ... ERROR in nls(). For further information see help(gcFitModel)
</code></pre>

<pre><code>## 
## 
## = 49. growth curve =================================
## ----------------------------------------------------
## --&gt; Try to fit model logistic
</code></pre>

<pre><code>## ....... OK
</code></pre>

<pre><code>## --&gt; Try to fit model richards
</code></pre>

<pre><code>## ....... ERROR in nls(). For further information see help(gcFitModel)
</code></pre>

<pre><code>## --&gt; Try to fit model gompertz
</code></pre>

<pre><code>## ....... OK
</code></pre>

<pre><code>## --&gt; Try to fit model gompertz.exp
</code></pre>

<pre><code>## ... ERROR in nls(). For further information see help(gcFitModel)
</code></pre>

<pre><code>## 
## 
## = 50. growth curve =================================
## ----------------------------------------------------
## --&gt; Try to fit model logistic
</code></pre>

<pre><code>## ....... OK
</code></pre>

<pre><code>## --&gt; Try to fit model richards
</code></pre>

<pre><code>## ....... ERROR in nls(). For further information see help(gcFitModel)
</code></pre>

<pre><code>## --&gt; Try to fit model gompertz
</code></pre>

<pre><code>## ....... OK
</code></pre>

<pre><code>## --&gt; Try to fit model gompertz.exp
</code></pre>

<pre><code>## ... ERROR in nls(). For further information see help(gcFitModel)
</code></pre>

<pre><code>## 
## 
## = 51. growth curve =================================
## ----------------------------------------------------
## --&gt; Try to fit model logistic
</code></pre>

<pre><code>## ....... OK
</code></pre>

<pre><code>## --&gt; Try to fit model richards
</code></pre>

<pre><code>## ....... ERROR in nls(). For further information see help(gcFitModel)
</code></pre>

<pre><code>## --&gt; Try to fit model gompertz
</code></pre>

<pre><code>## ....... OK
</code></pre>

<pre><code>## --&gt; Try to fit model gompertz.exp
</code></pre>

<pre><code>## ... ERROR in nls(). For further information see help(gcFitModel)
</code></pre>

<pre><code>## 
## 
## = 52. growth curve =================================
## ----------------------------------------------------
## --&gt; Try to fit model logistic
</code></pre>

<pre><code>## ....... OK
</code></pre>

<pre><code>## --&gt; Try to fit model richards
</code></pre>

<pre><code>## ....... ERROR in nls(). For further information see help(gcFitModel)
</code></pre>

<pre><code>## --&gt; Try to fit model gompertz
</code></pre>

<pre><code>## ....... OK
</code></pre>

<pre><code>## --&gt; Try to fit model gompertz.exp
</code></pre>

<pre><code>## ... ERROR in nls(). For further information see help(gcFitModel)
</code></pre>

<pre><code>## 
## 
## = 53. growth curve =================================
## ----------------------------------------------------
## --&gt; Try to fit model logistic
</code></pre>

<pre><code>## ....... OK
</code></pre>

<pre><code>## --&gt; Try to fit model richards
</code></pre>

<pre><code>## ....... ERROR in nls(). For further information see help(gcFitModel)
</code></pre>

<pre><code>## --&gt; Try to fit model gompertz
</code></pre>

<pre><code>## ....... OK
</code></pre>

<pre><code>## --&gt; Try to fit model gompertz.exp
</code></pre>

<pre><code>## ... ERROR in nls(). For further information see help(gcFitModel)
</code></pre>

<pre><code>## 
## 
## = 54. growth curve =================================
## ----------------------------------------------------
## --&gt; Try to fit model logistic
</code></pre>

<pre><code>## ....... OK
</code></pre>

<pre><code>## --&gt; Try to fit model richards
</code></pre>

<pre><code>## ....... ERROR in nls(). For further information see help(gcFitModel)
</code></pre>

<pre><code>## --&gt; Try to fit model gompertz
</code></pre>

<pre><code>## ....... OK
</code></pre>

<pre><code>## --&gt; Try to fit model gompertz.exp
</code></pre>

<pre><code>## ... ERROR in nls(). For further information see help(gcFitModel)
</code></pre>

<pre><code>## 
## 
## = 55. growth curve =================================
## ----------------------------------------------------
## --&gt; Try to fit model logistic
</code></pre>

<pre><code>## ....... OK
</code></pre>

<pre><code>## --&gt; Try to fit model richards
</code></pre>

<pre><code>## ....... ERROR in nls(). For further information see help(gcFitModel)
</code></pre>

<pre><code>## --&gt; Try to fit model gompertz
</code></pre>

<pre><code>## ....... OK
</code></pre>

<pre><code>## --&gt; Try to fit model gompertz.exp
</code></pre>

<pre><code>## ... ERROR in nls(). For further information see help(gcFitModel)
</code></pre>

<pre><code>## 
## 
## = 56. growth curve =================================
## ----------------------------------------------------
## --&gt; Try to fit model logistic
</code></pre>

<pre><code>## ....... OK
</code></pre>

<pre><code>## --&gt; Try to fit model richards
</code></pre>

<pre><code>## ....... ERROR in nls(). For further information see help(gcFitModel)
</code></pre>

<pre><code>## --&gt; Try to fit model gompertz
</code></pre>

<pre><code>## ....... OK
</code></pre>

<pre><code>## --&gt; Try to fit model gompertz.exp
</code></pre>

<pre><code>## ... ERROR in nls(). For further information see help(gcFitModel)
</code></pre>

<pre><code>## 
## 
## = 57. growth curve =================================
## ----------------------------------------------------
## --&gt; Try to fit model logistic
</code></pre>

<pre><code>## ....... OK
</code></pre>

<pre><code>## --&gt; Try to fit model richards
</code></pre>

<pre><code>## ....... ERROR in nls(). For further information see help(gcFitModel)
</code></pre>

<pre><code>## --&gt; Try to fit model gompertz
</code></pre>

<pre><code>## ....... OK
</code></pre>

<pre><code>## --&gt; Try to fit model gompertz.exp
</code></pre>

<pre><code>## ... ERROR in nls(). For further information see help(gcFitModel)
</code></pre>

<pre><code>## 
## 
## = 58. growth curve =================================
## ----------------------------------------------------
## --&gt; Try to fit model logistic
</code></pre>

<pre><code>## ....... OK
</code></pre>

<pre><code>## --&gt; Try to fit model richards
</code></pre>

<pre><code>## ....... ERROR in nls(). For further information see help(gcFitModel)
</code></pre>

<pre><code>## --&gt; Try to fit model gompertz
</code></pre>

<pre><code>## ....... OK
</code></pre>

<pre><code>## --&gt; Try to fit model gompertz.exp
</code></pre>

<pre><code>## ... ERROR in nls(). For further information see help(gcFitModel)
</code></pre>

<pre><code>## 
## 
## = 59. growth curve =================================
## ----------------------------------------------------
## --&gt; Try to fit model logistic
</code></pre>

<pre><code>## ....... OK
</code></pre>

<pre><code>## --&gt; Try to fit model richards
</code></pre>

<pre><code>## ....... ERROR in nls(). For further information see help(gcFitModel)
</code></pre>

<pre><code>## --&gt; Try to fit model gompertz
</code></pre>

<pre><code>## ....... OK
</code></pre>

<pre><code>## --&gt; Try to fit model gompertz.exp
</code></pre>

<pre><code>## ... ERROR in nls(). For further information see help(gcFitModel)
</code></pre>

<pre><code>## 
## 
## = 60. growth curve =================================
## ----------------------------------------------------
## --&gt; Try to fit model logistic
</code></pre>

<pre><code>## ....... OK
</code></pre>

<pre><code>## --&gt; Try to fit model richards
</code></pre>

<pre><code>## ....... ERROR in nls(). For further information see help(gcFitModel)
</code></pre>

<pre><code>## --&gt; Try to fit model gompertz
</code></pre>

<pre><code>## ....... OK
</code></pre>

<pre><code>## --&gt; Try to fit model gompertz.exp
</code></pre>

<pre><code>## ... ERROR in nls(). For further information see help(gcFitModel)
</code></pre>

<pre><code>## 
## 
## = 61. growth curve =================================
## ----------------------------------------------------
## --&gt; Try to fit model logistic
</code></pre>

<pre><code>## ....... OK
</code></pre>

<pre><code>## --&gt; Try to fit model richards
</code></pre>

<pre><code>## ....... ERROR in nls(). For further information see help(gcFitModel)
</code></pre>

<pre><code>## --&gt; Try to fit model gompertz
</code></pre>

<pre><code>## ....... OK
</code></pre>

<pre><code>## --&gt; Try to fit model gompertz.exp
</code></pre>

<pre><code>## ... ERROR in nls(). For further information see help(gcFitModel)
</code></pre>

<pre><code>## 
## 
## = 62. growth curve =================================
## ----------------------------------------------------
## --&gt; Try to fit model logistic
</code></pre>

<pre><code>## ....... OK
</code></pre>

<pre><code>## --&gt; Try to fit model richards
</code></pre>

<pre><code>## ....... ERROR in nls(). For further information see help(gcFitModel)
</code></pre>

<pre><code>## --&gt; Try to fit model gompertz
</code></pre>

<pre><code>## ....... OK
</code></pre>

<pre><code>## --&gt; Try to fit model gompertz.exp
</code></pre>

<pre><code>## ... ERROR in nls(). For further information see help(gcFitModel)
</code></pre>

<pre><code>## 
## 
## = 63. growth curve =================================
## ----------------------------------------------------
## --&gt; Try to fit model logistic
</code></pre>

<pre><code>## ....... OK
</code></pre>

<pre><code>## --&gt; Try to fit model richards
</code></pre>

<pre><code>## ....... ERROR in nls(). For further information see help(gcFitModel)
</code></pre>

<pre><code>## --&gt; Try to fit model gompertz
</code></pre>

<pre><code>## ....... OK
</code></pre>

<pre><code>## --&gt; Try to fit model gompertz.exp
</code></pre>

<pre><code>## ... ERROR in nls(). For further information see help(gcFitModel)
</code></pre>

<pre><code>## 
## 
## = 64. growth curve =================================
## ----------------------------------------------------
## --&gt; Try to fit model logistic
</code></pre>

<pre><code>## ....... OK
</code></pre>

<pre><code>## --&gt; Try to fit model richards
</code></pre>

<pre><code>## ....... ERROR in nls(). For further information see help(gcFitModel)
</code></pre>

<pre><code>## --&gt; Try to fit model gompertz
</code></pre>

<pre><code>## ....... OK
</code></pre>

<pre><code>## --&gt; Try to fit model gompertz.exp
</code></pre>

<pre><code>## ... ERROR in nls(). For further information see help(gcFitModel)
</code></pre>

<pre><code>## 
## 
## = 65. growth curve =================================
## ----------------------------------------------------
## --&gt; Try to fit model logistic
</code></pre>

<pre><code>## ....... OK
</code></pre>

<pre><code>## --&gt; Try to fit model richards
</code></pre>

<pre><code>## ....... ERROR in nls(). For further information see help(gcFitModel)
</code></pre>

<pre><code>## --&gt; Try to fit model gompertz
</code></pre>

<pre><code>## ....... OK
</code></pre>

<pre><code>## --&gt; Try to fit model gompertz.exp
</code></pre>

<pre><code>## ... ERROR in nls(). For further information see help(gcFitModel)
</code></pre>

<pre><code>## 
## 
## = 66. growth curve =================================
## ----------------------------------------------------
## --&gt; Try to fit model logistic
</code></pre>

<pre><code>## ....... OK
</code></pre>

<pre><code>## --&gt; Try to fit model richards
</code></pre>

<pre><code>## ....... ERROR in nls(). For further information see help(gcFitModel)
</code></pre>

<pre><code>## --&gt; Try to fit model gompertz
</code></pre>

<pre><code>## ....... OK
</code></pre>

<pre><code>## --&gt; Try to fit model gompertz.exp
</code></pre>

<pre><code>## ... ERROR in nls(). For further information see help(gcFitModel)
</code></pre>

<pre><code>## 
## 
## = 67. growth curve =================================
## ----------------------------------------------------
## --&gt; Try to fit model logistic
</code></pre>

<pre><code>## ....... OK
</code></pre>

<pre><code>## --&gt; Try to fit model richards
</code></pre>

<pre><code>## ....... ERROR in nls(). For further information see help(gcFitModel)
</code></pre>

<pre><code>## --&gt; Try to fit model gompertz
</code></pre>

<pre><code>## ....... OK
</code></pre>

<pre><code>## --&gt; Try to fit model gompertz.exp
</code></pre>

<pre><code>## ... ERROR in nls(). For further information see help(gcFitModel)
</code></pre>

<pre><code>## 
## 
## = 68. growth curve =================================
## ----------------------------------------------------
## --&gt; Try to fit model logistic
</code></pre>

<pre><code>## ....... OK
</code></pre>

<pre><code>## --&gt; Try to fit model richards
</code></pre>

<pre><code>## ....... ERROR in nls(). For further information see help(gcFitModel)
</code></pre>

<pre><code>## --&gt; Try to fit model gompertz
</code></pre>

<pre><code>## ....... OK
</code></pre>

<pre><code>## --&gt; Try to fit model gompertz.exp
</code></pre>

<pre><code>## ... ERROR in nls(). For further information see help(gcFitModel)
</code></pre>

<pre><code>## 
## 
## = 69. growth curve =================================
## ----------------------------------------------------
## --&gt; Try to fit model logistic
</code></pre>

<pre><code>## ....... OK
</code></pre>

<pre><code>## --&gt; Try to fit model richards
</code></pre>

<pre><code>## ....... ERROR in nls(). For further information see help(gcFitModel)
</code></pre>

<pre><code>## --&gt; Try to fit model gompertz
</code></pre>

<pre><code>## ....... OK
</code></pre>

<pre><code>## --&gt; Try to fit model gompertz.exp
</code></pre>

<pre><code>## ... ERROR in nls(). For further information see help(gcFitModel)
</code></pre>

<pre><code>## 
## 
## = 70. growth curve =================================
## ----------------------------------------------------
## --&gt; Try to fit model logistic
</code></pre>

<pre><code>## ....... OK
</code></pre>

<pre><code>## --&gt; Try to fit model richards
</code></pre>

<pre><code>## ....... ERROR in nls(). For further information see help(gcFitModel)
</code></pre>

<pre><code>## --&gt; Try to fit model gompertz
</code></pre>

<pre><code>## ....... OK
</code></pre>

<pre><code>## --&gt; Try to fit model gompertz.exp
</code></pre>

<pre><code>## ... ERROR in nls(). For further information see help(gcFitModel)
</code></pre>

<pre><code>## 
## 
## = 71. growth curve =================================
## ----------------------------------------------------
## --&gt; Try to fit model logistic
</code></pre>

<pre><code>## ....... ERROR in nls(). For further information see help(gcFitModel)
</code></pre>

<pre><code>## --&gt; Try to fit model richards
</code></pre>

<pre><code>## ....... ERROR in nls(). For further information see help(gcFitModel)
</code></pre>

<pre><code>## --&gt; Try to fit model gompertz
</code></pre>

<pre><code>## ....... ERROR in nls(). For further information see help(gcFitModel)
</code></pre>

<pre><code>## --&gt; Try to fit model gompertz.exp
</code></pre>

<pre><code>## ... ERROR in nls(). For further information see help(gcFitModel)
</code></pre>

<pre><code>## Warning in grofit::gcFitModel(acttime, actwell, gcID, control): gcFitModel:
## Unable to fit this curve parametrically!
</code></pre>

<pre><code>## 
## 
## = 72. growth curve =================================
## ----------------------------------------------------
## --&gt; Try to fit model logistic
</code></pre>

<pre><code>## ....... OK
</code></pre>

<pre><code>## --&gt; Try to fit model richards
</code></pre>

<pre><code>## ....... ERROR in nls(). For further information see help(gcFitModel)
</code></pre>

<pre><code>## --&gt; Try to fit model gompertz
</code></pre>

<pre><code>## ....... OK
</code></pre>

<pre><code>## --&gt; Try to fit model gompertz.exp
</code></pre>

<pre><code>## ... ERROR in nls(). For further information see help(gcFitModel)
</code></pre>

<pre><code>## 
## 
## = 73. growth curve =================================
## ----------------------------------------------------
## --&gt; Try to fit model logistic
</code></pre>

<pre><code>## ....... OK
</code></pre>

<pre><code>## --&gt; Try to fit model richards
</code></pre>

<pre><code>## ....... ERROR in nls(). For further information see help(gcFitModel)
</code></pre>

<pre><code>## --&gt; Try to fit model gompertz
</code></pre>

<pre><code>## ....... OK
</code></pre>

<pre><code>## --&gt; Try to fit model gompertz.exp
</code></pre>

<pre><code>## ... ERROR in nls(). For further information see help(gcFitModel)
</code></pre>

<pre><code>## 
## 
## = 74. growth curve =================================
## ----------------------------------------------------
## --&gt; Try to fit model logistic
</code></pre>

<pre><code>## ....... OK
</code></pre>

<pre><code>## --&gt; Try to fit model richards
</code></pre>

<pre><code>## ....... ERROR in nls(). For further information see help(gcFitModel)
</code></pre>

<pre><code>## --&gt; Try to fit model gompertz
</code></pre>

<pre><code>## ....... OK
</code></pre>

<pre><code>## --&gt; Try to fit model gompertz.exp
</code></pre>

<pre><code>## ... ERROR in nls(). For further information see help(gcFitModel)
</code></pre>

<pre><code>## 
## 
## = 75. growth curve =================================
## ----------------------------------------------------
## --&gt; Try to fit model logistic
</code></pre>

<pre><code>## ....... OK
</code></pre>

<pre><code>## --&gt; Try to fit model richards
</code></pre>

<pre><code>## ....... ERROR in nls(). For further information see help(gcFitModel)
</code></pre>

<pre><code>## --&gt; Try to fit model gompertz
</code></pre>

<pre><code>## ....... OK
</code></pre>

<pre><code>## --&gt; Try to fit model gompertz.exp
</code></pre>

<pre><code>## ... ERROR in nls(). For further information see help(gcFitModel)
</code></pre>

<pre><code>## 
## 
## = 76. growth curve =================================
## ----------------------------------------------------
## --&gt; Try to fit model logistic
</code></pre>

<pre><code>## ....... OK
</code></pre>

<pre><code>## --&gt; Try to fit model richards
</code></pre>

<pre><code>## ....... ERROR in nls(). For further information see help(gcFitModel)
</code></pre>

<pre><code>## --&gt; Try to fit model gompertz
</code></pre>

<pre><code>## ....... OK
</code></pre>

<pre><code>## --&gt; Try to fit model gompertz.exp
</code></pre>

<pre><code>## ... ERROR in nls(). For further information see help(gcFitModel)
</code></pre>

<pre><code>## 
## 
## = 77. growth curve =================================
## ----------------------------------------------------
## --&gt; Try to fit model logistic
</code></pre>

<pre><code>## ....... OK
</code></pre>

<pre><code>## --&gt; Try to fit model richards
</code></pre>

<pre><code>## ....... ERROR in nls(). For further information see help(gcFitModel)
</code></pre>

<pre><code>## --&gt; Try to fit model gompertz
</code></pre>

<pre><code>## ....... OK
</code></pre>

<pre><code>## --&gt; Try to fit model gompertz.exp
</code></pre>

<pre><code>## ... ERROR in nls(). For further information see help(gcFitModel)
</code></pre>

<pre><code class="r">dev.off()
</code></pre>

<pre><code>## png 
##   2
</code></pre>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-28" style="background:;">
  <hgroup>
    <h3>Use Package <code>grofit</code> to Fit Growth Data</h3>
  </hgroup>
  <article data-timings="">
    <pre><code class="r">table &lt;- grofitGetParameters(fits, p=c(&quot;AddId&quot;,&quot;mu.spline&quot;))
boxplot(table[,&quot;mu.spline&quot;] ~ table[,&quot;AddId&quot;], las=2, ylim=c(1.9e-4,5e-4))
</code></pre>

<p><img src="assets/fig/unnamed-chunk-21-1.png" alt="plot of chunk unnamed-chunk-21"></p>

<p><oq> 
Try different well groups instead of <code>table[,&quot;AddId&quot;]</code>.<br/>
Are the fits Ok? How to get more information on the fits?</oq></p>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-29" style="background:;">
  <article data-timings="">
    
<div style='float:left;width:48%;' class='centered'>
  <h3>Gene Expression</h3>

<ul>
<li>Measured: total fluorescence per well</li>
<li>Wanted: protein level per cell!</li>
</ul>

<p>Assumptions:</p>

<ol>
<li>fluorescence is linear with protein level</li>
<li>OD is linear with cell number</li>
</ol>

<p><oq>\( \Rightarrow \) Fluor./OD  &Tilde; proteins/cell</oq></p>

</div>
<div style='float:right;width:48%;'>
  <p><img src="assets/img/ecoli_20141014.png" height="190">
<img src="assets/img/Ecoli_20161014_OD_grofit_A8.png" height="190">
<img src="assets/img/Ecoli_20161014_OD_growthrates.png" height="190"></p>

</div>
  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-30" style="background:;">
  <hgroup>
    <h3>Gene Expression: Normalized Fluorescence - FL/OD</h3>
  </hgroup>
  <article data-timings="">
    <pre><code class="r">viewGroups(data, groups=groups, groups2=groups2, xid=&quot;OD&quot;, lwd.orig=.5, verb=FALSE)
</code></pre>

<p><img src="assets/fig/unnamed-chunk-22-1.png" alt="plot of chunk unnamed-chunk-22"></p>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-31" style="background:;">
  <hgroup>
    <h3>Gene Expression: Normalized Fluorescence  - FL/OD</h3>
  </hgroup>
  <article data-timings="">
    <pre><code class="r">fl &lt;- getData(data,&quot;mVenus&quot;)
od &lt;- getData(data,&quot;OD&quot;)
data &lt;- addData(data, ID=&quot;mV/OD&quot;, dat=fl/od, col=&quot;#0095FF&quot;)
viewGroups(data, groups=groups, groups2=groups2, 
           dids=c(&quot;OD&quot;,&quot;mV/OD&quot;),ylims=list(&quot;mV/OD&quot;=c(0,1e4)),emphasize.mean=T,verb=F)
</code></pre>

<p><img src="assets/fig/unnamed-chunk-23-1.png" alt="plot of chunk unnamed-chunk-23"></p>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-32" style="background:;">
  <hgroup>
    <h3>Gene Expression: Normalized Fluorescence - FL/OD</h3>
  </hgroup>
  <article data-timings="">
    <pre><code class="r">viewGroups(data, groups=groups, groups2=groups2, dids=&quot;mV/OD&quot;, xid=&quot;OD&quot;, lwd.orig=.5, 
           ylim=c(0,1e4), g2.legend=FALSE, verb=FALSE)
</code></pre>

<p><img src="assets/fig/unnamed-chunk-24-1.png" alt="plot of chunk unnamed-chunk-24"></p>

<p><oq>What could the slope \(\frac{mV}{OD} \sim OD \) (where it&#39;s linear) 
mean?<br/>Can we use it?</oq></p>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-33" style="background:;">
  <hgroup>
    <h3>Gene Expression: Normalized Fluorescence - Interpolate to OD</h3>
  </hgroup>
  <article data-timings="">
    <pre><code class="r">od.data &lt;- interpolatePlateData(data,&quot;OD&quot;)
viewGroups(od.data, groups=groups[2], groups2=groups2,xlim=c(.01,.15),
           ylims=list(&quot;mV/OD&quot;=c(0,1e4)),dids=&quot;mV/OD&quot;, show.ci=F,lwd.orig=0, verb=F)
</code></pre>

<p><img src="assets/fig/unnamed-chunk-25-1.png" alt="plot of chunk unnamed-chunk-25"></p>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-34" style="background:;">
  <hgroup>
    <h3>Gene Expression: The Fold Change</h3>
  </hgroup>
  <article data-timings="">
    <pre><code class="r">flod &lt;- getData(od.data,&quot;mV/OD&quot;)
uninduced &lt;- rowMeans(flod[, groups2[[&quot;pUC18mV_Inj:NA&quot;]]],na.rm=TRUE)
od.data &lt;- addData(od.data, ID=&quot;mV/OD/uninduced&quot;, dat=flod/uninduced, col=&quot;#AAAA00&quot;)
</code></pre>

<p><img src="assets/fig/unnamed-chunk-26-1.png" alt="plot of chunk unnamed-chunk-26"></p>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-35" style="background:;">
  <hgroup>
    <h3>Gene Expression: Data Normalization</h3>
  </hgroup>
  <article data-timings="">
    <p><img src="assets/fig/unnamed-chunk-27-1.png" alt="plot of chunk unnamed-chunk-27">
<img src="assets/fig/unnamed-chunk-28-1.png" alt="plot of chunk unnamed-chunk-28"></p>

<p><oq>What&#39;s wrong with <code>mV/OD/uninduced</code> vs. <code>time</code> w/o interpolation?</oq></p>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-36" style="background:;">
  <hgroup>
    <h3>Gene Expression: Normalized Fluorescence - Result</h3>
  </hgroup>
  <article data-timings="">
    <pre><code class="r">results &lt;- boxData(od.data, did=&quot;mV/OD/uninduced&quot;, rng=.08, groups=groups2, type=&quot;bar&quot;)
</code></pre>

<p><img src="assets/fig/unnamed-chunk-29-1.png" alt="plot of chunk unnamed-chunk-29"></p>

<p>Extracts values in given range; plots by groups, and returns
values for each well. </p>

<p><oq>What are the error bars?<br/>
See ?cutData on how to obtain values in 
a given range directly.<br/>Or try yourself with base R.</oq></p>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-37" style="background:;">
  <hgroup>
    <h3>Gene Expression: Normalized Fluorescence - Result</h3>
  </hgroup>
  <article data-timings="">
    <pre><code class="r">library(platexpress)
head(results)
</code></pre>

<pre><code>##   well     group mV/OD/uninduced@0.08038
## 1   A1 WT_Inj:NA               0.6856046
## 2   B1 WT_Inj:NA               0.6510172
## 3   C1 WT_Inj:NA               0.6200460
## 4   D1 WT_Inj:NA               0.7815149
## 5   E1 WT_Inj:NA               0.6994579
## 6   F1 WT_Inj:NA               0.6037127
</code></pre>

<p><oq>Calculate confidence intervals and error bars yourself.</oq></p>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-38" style="background:;">
  <hgroup>
    <h3>Gene Expression - Summary</h3>
  </hgroup>
  <article data-timings="">
    <ol>
<li>Growth rate \( \mu \) matters!</li>
<li>Fluorescence per OD as a measure of proteins per cell.</li>
<li>Interpolate data to a common OD.

<ul>
<li>Note noise accumulation!</li>
</ul></li>
<li>Calculate fold-ratio to control.

<ul>
<li>Be careful; what is your control?</li>
</ul></li>
<li>Get data at a given OD, or search maximum, etc.</li>
</ol>

<p><oq>What other measures would be interesting or could be
required for a quantitative model?</oq></p>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="centertext" id="slide-39" style="background:;">
  <hgroup>
    <h2>From Wells to Cells</h2>
  </hgroup>
  <article data-timings="">
    <h3>Conversion of Carbon&amp;Energy Source S to biomass X:</h3>

<p><br/>
S \(\rightarrow\) y X + (1-y) P</p>

<h3>Catalyzed by X:</h3>

<p>\[
\begin{equation}
  \label{apeq:monod}
  \begin{aligned}
    \frac{\text{d}X}{\text{d}t} &= \mu X\\
    \frac{\text{d}S}{\text{d}t} &= - \frac{1}{y} \mu X \\
    \mu & =\mu_{max}\frac{S}{S+K}
    %\mu &= f(S)
  \end{aligned}
\end{equation}
\]</p>

<p><br/>
<oq>How can we estimate the yield \(y\) ?</oq></p>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="codefont" id="slide-40" style="background:;">
  <hgroup>
    <h2>From Wells to Cells</h2>
  </hgroup>
  <article data-timings="">
    <h3>Fluorescent Protein F, total culture concentration:</h3>

<p>\[
\begin{equation}
  \label{eq:Protein}
  \begin{aligned}
%  \frac{\text{d} F}{\text{d} t} = k X V_c - d F
    \frac{\text{d}X}{\text{d}t} &= \mu X\\
  \frac{\text{d} F}{\text{d} t} &= K(?) - d F
  \end{aligned}
\end{equation}
\]</p>

<h3>Intracellular concentration:</h3>

<p>\[
\begin{equation}
f = \frac{F}{X V_c}
\end{equation}
\]</p>

<p><br/> <oq>What is \(K\) ?</br>How can we map \(F\) from total culture
to intracellular concentration \(f\) ?<br/> Calculate \(\frac{\text{d}
f}{\text{d} t}\).</oq></p>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="codefont" id="slide-41" style="background:;">
  <hgroup>
    <h2>From Wells to Cells</h2>
  </hgroup>
  <article data-timings="">
    <p>\[
\begin{equation}
  \begin{aligned}
    f =& \frac{F}{X V_c}\\
    \dot{f} =& \dot{F}\frac{1}{X V_c} - \dot{X}\frac{F}{X}\frac{1}{X V_c}\\
            =& \dot{F}\frac{1}{v_f} - \dot{X}\frac{F}{X}\frac{1}{v_f}\\
            =& (K - d\,f\,v_f)\frac{1}{v_f} - \mu X \frac{f v_f}{X} \frac{1}{v_f}\\
            =& \frac{K}{v_f} - d\,f - \mu\,f\\
            =& \frac{K}{v_f} - (d+\mu)f
  \end{aligned}
\end{equation}
\]</p>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="codefont" id="slide-42" style="background:;">
  <hgroup>
    <h2>From Wells to Cells</h2>
  </hgroup>
  <article data-timings="">
    <p>\[
\begin{equation}
  \label{apeq:wellmodel}
  \begin{aligned}
    \frac{\text{d}X}{\text{d}t} &= \mu X\\
    \frac{\text{d}S}{\text{d}t} &= - \frac{1}{y} \mu X\\
    \frac{\text{d}f}{\text{d}t} &= k - (d+\mu)f
  \end{aligned}
\end{equation}
\]</p>

<ul>
<li><oq>Add activation of \(f\) production by an inducer and
modulation by a riboswitch \(r\).</oq></li>
<li><oq>Add expression of both from a plasmid
(which itself increases exponentially within cells), where the plasmid
copy number \(p\) can be accounted for</oq>. </li>
<li><oq>And how can we account for
the effect of induced gene expression on growth rate?</oq></li>
</ul>

  </article>
  <!-- Presenter Notes -->
</slide>

    <slide class="backdrop"></slide>
  </slides>
  <div class="pagination pagination-small" id='io2012-ptoc' style="display:none;">
    <ul>
      <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=1 title='NA'>
         1
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=2 title='NA'>
         2
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=3 title='Installing R Packages from <code>cran</code>, <code>bioconductor</code> &amp; <code>github</code>'>
         3
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=4 title='How to use a new R package?'>
         4
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=5 title='Why Growth Rates?'>
         5
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=6 title='Why Growth Rates?'>
         6
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=7 title='Why Growth Rates?'>
         7
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=8 title='NA'>
         8
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=9 title='Growth &amp; Gene Expression in <em>E. coli</em> : exponential growth'>
         9
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=10 title='Growth &amp; Gene Expression in <em>E. coli</em> : growth rate'>
         10
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=11 title='Growth &amp; Gene Expression in <em>E. coli</em> : doubling time'>
         11
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=12 title='Growth &amp; Gene Expression in <em>E. coli</em> : growth rate'>
         12
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=13 title='Load Your Data'>
         13
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=14 title='NA'>
         14
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=15 title='NA'>
         15
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=16 title='NA'>
         16
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=17 title='NA'>
         17
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=18 title='NA'>
         18
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=19 title='NA'>
         19
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=20 title='NA'>
         20
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=21 title='NA'>
         21
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=22 title='NA'>
         22
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=23 title='NA'>
         23
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=24 title='Fitting Growth Models'>
         24
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=25 title='Prepare Data: blanks, cuts, etc.'>
         25
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=26 title='Get Replicate Groups'>
         26
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=27 title='Use Package <code>grofit</code> to Fit Growth Data'>
         27
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=28 title='Use Package <code>grofit</code> to Fit Growth Data'>
         28
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=29 title='NA'>
         29
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=30 title='Gene Expression: Normalized Fluorescence - FL/OD'>
         30
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=31 title='Gene Expression: Normalized Fluorescence  - FL/OD'>
         31
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=32 title='Gene Expression: Normalized Fluorescence - FL/OD'>
         32
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=33 title='Gene Expression: Normalized Fluorescence - Interpolate to OD'>
         33
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=34 title='Gene Expression: The Fold Change'>
         34
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=35 title='Gene Expression: Data Normalization'>
         35
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=36 title='Gene Expression: Normalized Fluorescence - Result'>
         36
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=37 title='Gene Expression: Normalized Fluorescence - Result'>
         37
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=38 title='Gene Expression - Summary'>
         38
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=39 title='From Wells to Cells'>
         39
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=40 title='From Wells to Cells'>
         40
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=41 title='From Wells to Cells'>
         41
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=42 title='From Wells to Cells'>
         42
      </a>
    </li>
  </ul>
  </div>  <!--[if IE]>
    <script 
      src="http://ajax.googleapis.com/ajax/libs/chrome-frame/1/CFInstall.min.js">  
    </script>
    <script>CFInstall.check({mode: 'overlay'});</script>
  <![endif]-->
</body>
  <!-- Load Javascripts for Widgets -->
  
  <!-- MathJax: Fall back to local if CDN offline but local image fonts are not supported (saves >100MB) -->
  <script type="text/x-mathjax-config">
    MathJax.Hub.Config({
      tex2jax: {
        inlineMath: [['$','$'], ['\\(','\\)']],
        processEscapes: true
      }
    });
  </script>
  <script type="text/javascript" src="http://cdn.mathjax.org/mathjax/2.0-latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
  <!-- <script src="https://c328740.ssl.cf1.rackcdn.com/mathjax/2.0-latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML">
  </script> -->
  <script>window.MathJax || document.write('<script type="text/x-mathjax-config">MathJax.Hub.Config({"HTML-CSS":{imageFont:null}});<\/script><script src="libraries/widgets/mathjax/MathJax.js?config=TeX-AMS-MML_HTMLorMML"><\/script>')
</script>
<!-- LOAD HIGHLIGHTER JS FILES -->
  <script src="libraries/highlighters/highlight.js/highlight.pack.js"></script>
  <script>hljs.initHighlightingOnLoad();</script>
  <!-- DONE LOADING HIGHLIGHTER JS FILES -->
   
  </html>